### PR TITLE
CI workflow to validate webhook templates

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Add CI workflow to validate webhook templates changes on pull requests.

#### Changes

- Add `validate.go` program that loads webhook templates from a local directory and validates the `TemplateStore` implementation
- Add GitHub workflow for running validate on each pull request and pushes to master. Binary and go modules are cached so workflow only takes a couple of seconds.
- Fix TagoIO and Ubidots webhook templates caught by the tool --> there is a mismatch on their ID vs the filename.

#### Notes for Reviewers

- The webhook template ID changes should not affect users that setup their integrations from the Console, since the console only uses the `ListTemplates` RPC.
- Changed the webhook template IDs to match the ones that existing integrations will already have, to avoid breaking something by accident. 

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible, they don't break existing deployments.
- [X] Testing: The changes are tested.
- [ ] Documentation: Relevant documentation is added or updated.

### CODE REVIEW

**Review by nejraselimovic — APPROVED** [review]

**Review by htdvisser — CHANGES_REQUESTED** [review]
I would prefer to do the imports the other way around; instead of having `lorawan-stack` as a dependency of `lorawan-webhook-templates`, maybe it would be better to define the structs in `lorawan-webhook-templates` and have `lorawan-stack` import those.

Also: Please see my message on Slack.

**Comment by adriansmares** [comment]
> I would prefer to do the imports the other way around; instead of having `lorawan-stack` as a dependency of `lorawan-webhook-templates`, maybe it would be better to define the structs in `lorawan-webhook-templates` and have `lorawan-stack` import those.
> 
> Also: Please see my message on Slack.

I initially wanted to just press approve, but I actually agree with this suggestion after giving it some thought:
- No cyclical dependencies
- The variant in this PR also in a sense tests the store implementation as well, which is not ideal (albeit useful in a sense)
- Checking if the YAML serialization / de-serialization passes suffices for this class of bugs

A long term solution would be a YAML schema, but I can accept just testing the serialization and a backlog issue for now.

**Review by adriansmares — APPROVED** [review]
I would make the `schema` package top level, and make sure to use it in `lorawan-stack`.

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Initialize tool binary cache
+      id: tools-cache
+      uses: actions/cache@v2
+      with:
+        path: bin
+        key: ${{ runner.os }}-validate-${{ hashFiles('schema/**.go') }}
```
Comment (htdvisser) [inline .github/workflows/validate.yml:21]: Please also include `go.mod`/`go.sum` so that we can update dependencies

File: schema/cmd/validate.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	"go.thethings.network/lorawan-webhook-templates/schema"
+	"gopkg.in/yaml.v2"
+)
+
+// numWorkers is the number of worker goroutines to spawn for template validation.
+const numWorkers = 12
+
+func validateTemplate(baseDir, templateID string) error {
+	templateYml := filepath.Join(baseDir, templateID+".yml")
+	b, err := ioutil.ReadFile(templateYml)
```
Comment (htdvisser) [inline schema/cmd/validate.go:35]: Since Go 1.16 package `ioutil` is deprecated, please use `os.ReadFile` instead.

File: schema/cmd/validate.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+	go func() {
+		for _, templateID := range templateIDs {
+			templateCh <- templateID
+		}
+		close(templateCh)
+	}()
+
+	wg.Wait()
+	os.Exit(rc)
```
Comment (htdvisser) [inline schema/cmd/validate.go:102]: You never seem to modify `rc`, so it's always `0`.

File: schema/go.mod [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module go.thethings.network/lorawan-webhook-templates/schema
```
Comment (htdvisser) [inline schema/go.mod:1]: This is not found. Either add it to https://github.com/TheThingsNetwork/govanityurls/blob/master/vanity.yaml or use github.com/TheThingsIndustries/lorawan-webhook-templates.

File: schema/cmd/validate.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	templateCh := make(chan string)
+
+	rc := 0
+	for i := 0; i < numWorkers; i++ {
+		wg.Add(1)
+		go func() {
+			for templateID := range templateCh {
+				if err := validateTemplate(baseDir, templateID); err != nil {
+					fmt.Fprintf(os.Stderr, "Webhook template %s ERROR: %v\n", templateID, err)
+					rc = 1
```
Comment (htdvisser) [inline schema/cmd/validate.go:85]: The race detector probably won't like this

**Review by htdvisser — APPROVED** [review]

**Pull request merged**